packages/shared/lib/core/layer-2/utils/getCommonEvmTransactionData.ts

```diff
@@ -0,0 +1,18 @@
+import Web3 from 'web3'
+import { TxData } from '@ethereumjs/tx'
+import { ISC_MAGIC_CONTRACT_ADDRESS } from '@core/layer-2/constants'
+
+export async function getCommonTransactionData(provider: Web3, originAddress: string, data: string): Promise<TxData> {
+    const nonce = provider.utils.toHex(await provider.eth.getTransactionCount(originAddress))
+
+    const _gasPrice = await provider.eth.getGasPrice()
+    const gasPrice = '0x' + _gasPrice
+
+    const estimatedGas = await provider.eth.estimateGas({ from: originAddress, to: ISC_MAGIC_CONTRACT_ADDRESS, data })
+    const gasLimit = provider.utils.toHex(2 * estimatedGas) // Double to ensure we have enough gas
```

Comment:
We have a `GAS_MULTIPLIER` constant that we should use here, avoiding a random magic constant. Perhaps `GAS_LIMIT_MULTIPLIER` is slightly more explicit too?

Reply:
I will add `GAS_MULTIPLIER`, but we will need to make this whole gas setting dynamic in the future.